app/javascript/crayons/formElements/ColorPicker/ColorPicker.jsx

```diff
@@ -0,0 +1,69 @@
+import { h, Fragment } from 'preact';
+import PropTypes from 'prop-types';
+import { useState, useLayoutEffect } from 'preact/hooks';
+import { HexColorPicker, HexColorInput } from 'react-colorful';
+import { initializeDropdown } from '@utilities/dropdownUtils';
+import { ButtonNew as Button } from '@crayons';
+
+export const ColorPicker = ({
+  id,
+  labelText,
+  defaultValue,
+  showLabel = true,
+}) => {
+  // Ternary has been used here to guard against an empty string being passed as default value
+  const [color, setColor] = useState(defaultValue ? defaultValue : '#000');
+
+  const buttonId = `color-popover-btn-${id}`;
+  const popoverId = `color-popover-${id}`;
+
+  useLayoutEffect(() => {
+    initializeDropdown({
+      triggerElementId: buttonId,
+      dropdownContentId: popoverId,
+    });
+  }, [buttonId, popoverId]);
+
+  return (
+    <Fragment>
+      {showLabel && (
+        <label
+          for={id}
+          className={`crayons-field__label ${
```

Comment:
I think we should strip this component from that `<label>` element so we can have more granular control on how it's all is built. 

I think we should **not** keep labels/descriptions on this level - instead we should have generic <FormField> component that would be responsible for form field layout and generic structure. you would be able to specify there things like:

1. label (for `.crayons-field__label`),
2. description (for `.crayons-field__description`),
3. and finally what field you want (text input, textarea, checkbox, radio, select, etc).

So probably something like:

```
<FormField
  label="..."
  description="..."
  field={<ColorPicker ... />}
  ...
/>
```

...and the output would be like:

```
.crayons-field
    label.crayons-field__label
    p.crayons-field__description
    [form field]
```

This is obviously out of scope for this PR but just giving you context.

Reply:
Yeah I agree it's a bit odd to have the label within the component... the reason I did it is because both the button and the input need to reference the label text, so it makes it a bit "non standard" compared to other fields 🤔

I guess the alternative is that `ColorPicker` has a compulsory prop for the button's aria label prop, but it felt like duplication at the time since it's pretty much gonna be the same as the label for the input 🤷‍♀️ 

I don't mind separating them out like that though if it helps pave the way for this generic FormField component

Reply:
That would be nice and I think it would be reasonable do separate these out now since we already have this `crayons-field` element pretty much established so it's a matter of improving/polishing it a little till we start sayin "HEY YOU BETTER USE THIS BAD BOY SINCE IT'S OFFICIAL NOW"

And honestly I think we're very very close - we've just updated the design part of this component ([1](https://www.figma.com/file/gHpwugeu9jFGW4d9u3ReuR/Crayons---NEW?node-id=364%3A111), [2](https://www.figma.com/file/gHpwugeu9jFGW4d9u3ReuR/Crayons---NEW?node-id=387%3A319)) so we now just have to map it all in code (famous last words).

Reply:
I've split it out 🙂  We might just need to give it some special love when we implement the new stuff for `FormField`